src/modules/cpu/kernel/normalize_generic.hpp

```diff
@@ -0,0 +1,604 @@
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+void compute_diff_square_sum(Rpp32f &output, Rpp32f *input, Rpp32s inputStride, Rpp32s numElements, Rpp32f mean)
+{
+    const Rpp32s stride = 1;
+    if (numElements > 32)
+    {
+        Rpp32s currElements = numElements >> 1;
+        Rpp32f tmp1 = 0, tmp2 = 0;
+
+        // reduce first half and accumulate
+        compute_diff_square_sum(tmp1, input, stride, currElements, mean);
+
+        // reduce second half and accumulate
+        compute_diff_square_sum(tmp2, input + currElements * stride, stride, numElements - currElements, mean);
+
+        tmp1 += tmp2;
+        output += tmp1;
+    }
+    else
+    {
+        // reduce to a temporary
+        Rpp32f tmp = 0;
+        for (Rpp32s i = 0; i < numElements; i++)
+        {
+            Rpp32f curr = (input[i * stride] - mean);
+            auto curnew = curr * curr;
+            tmp += curnew;
+        }
+
+        // accumulate in target value
+        output += tmp;
+    }
+}
+
+void compute_sum(Rpp32f &output, Rpp32f *input, Rpp32s inputStride, Rpp32s numElements)
+{
+    const Rpp32s stride = 1;
+    if (numElements > 32)
+    {
+        Rpp32s currElements = numElements >> 1;
+        Rpp32f tmp1 = 0, tmp2 = 0;
+
+        // reduce first half and accumulate
+        compute_sum(tmp1, input, stride, currElements);
+
+        // reduce second half and accumulate
+        compute_sum(tmp2, input + currElements * stride, stride, numElements - currElements);
+
+        tmp1 += tmp2;
+        output += tmp1;
+    }
+    else
+    {
+        // reduce to a temporary
+        Rpp32f tmp = 0;
+        for (Rpp32s i = 0; i < numElements; i++)
+            tmp += input[i * stride];
+
+        // accumulate in target value
+        output += tmp;
+    }
+}
+
+Rpp32f rpp_rsqrt(Rpp32f x)
+{
+    // Use SSE intrinsic and one Newton-Raphson refinement step
+    // - faster and less hacky than the hack below.
+    __m128 X = _mm_set_ss(x);
+    __m128 tmp = _mm_rsqrt_ss(X);
+    Rpp32f y = _mm_cvtss_f32(tmp);
+    return y * (1.5f - x * 0.5f * y * y);
+}
+
+static void rpp_rsqrt_avx(Rpp32f *input, Rpp32s numElements, Rpp32f eps, Rpp32f rdiv, Rpp32f mul)
```

Comment:
I believe there is a funciton naming convention and placement location for rpp math funcionns in SSE/AVX inside the rpp_cpu_common or rpp_cpu_simd headers. Please place there.

Reply:
Placed it in rpp_cpu_simd.hpp but it lead to multiple linker errors like

/usr/bin/ld: src/modules/CMakeFiles/modules.dir/rppi_arithmetic_operations.cpp.o: in function `rpp_rsqrt_ps(float)':
rppi_arithmetic_operations.cpp:(.text+0x0): multiple definition of `rpp_rsqrt_ps(float)'; src/modules/CMakeFiles/modules.dir/rppi_advanced_augmentations.cpp.o:rppi_advanced_augmentations.cpp:(.text+0x0): first defined here